### DISCUSSION TITLE
보이스카웃 원칙을 지키는게 어떨까요?

### DISCUSSION THREAD

**Discussion by her0807** [woowacourse-teams/2022-ternoko#126] [Show and tell]
## 문제의식
안녕하세요. 행복한 월요일이네요.. ㅎㅎㅎ
코드를 짜다보니까 크게는 패키지 구조, fixtures , 작게는 final 과 같은 프로젝트 컨벤션 + 메서드 선언 순서 등,   개선할 점이 많이 보이네요.. 🤔 

어제 개발을 하나보니 문득, 제가 개선점을 발견 했음에도 불구하고 안치우고 있다는 걸 깨달았어요. (여러 이유에서.. 그랬겠지만..) 그래서 저희 팀 문화로 `보이스카웃 원칙`을 제안합니다!

## 보이스카웃 원칙이란?
간단히 요약하자면  `"떠날 때는 찾을 때 보다 캠프장을 더욱 깨끗이 하라." `라는 보이 스카웃의 전통적인 규칙을 인용한 내용인데요.
저희 팀이 지키면 좋겠다고 생각한 내용 아래와 같습니다. 

1. 본인이 작성한 코드에 대해서 레거시가 생기지 않게 주의해서 코드를 작성한다.
2. 코드를 작성하면서 개선점을 느꼈다면 주저하지 않고 본인이 리팩터링한다. 

## 참고
https://m.blog.naver.com/PostView.naver?isHttpsRedirect=true&blogId=nimbusob&logNo=161544641

**Comment by HyeonbinSa** [comment]
좋아요. 주인의식을 가지고 한번 더 체크해보도록하겠습니다.

**Comment by dongho108** [comment]
저도 저번에 이 원칙을 듣고나서 조금씩 인지하면서 개발하고 있어요
좋습니다.!!
